spec/models/routine_flow_spec.rb

```diff
@@ -40,15 +43,43 @@
     end
   end
 
-  describe '#complete!' do
+  describe '#complete_routine_flow!' do
     it 'should complete routine flow if step is the last step' do
       routine, routine_flow = start_routine_flow
 
       (routine.steps.count-1).times { routine_flow.take_next_flow_step! }
-      routine_flow.complete!
+      routine_flow.complete_routine_flow!
 
       expect(routine_flow).to be_complete
     end
+
+    it 'should update finished_at with current time' do
+      freeze_time
+      routine, routine_flow = start_routine_flow
+
+      (routine.steps.count-1).times { routine_flow.take_next_flow_step! }
+      routine_flow.complete_routine_flow!
+
+      expect(routine_flow.completed_at).to eq(Time.current)
+    end
+
+    it 'should update time_to_complete with the duration' do
+      freeze_time
+
+      start_time = Time.current - 7.minutes
+      finish_time = start_time + 7.minutes
+
+      travel_to start_time
+
+      routine, routine_flow = start_routine_flow
+
+      travel_to finish_time
+
+      (routine.steps.count-1).times { routine_flow.take_next_flow_step! }
```

Comment:
Swap these 2 lines - easier to understand test.

Reply:
Done, boss!